src/rust/scheduler/scheduler.rs

```diff
@@ -402,39 +421,82 @@ mod tests {
         Ok(())
     }
 
-    /// Tests if consecutive tasks are not assigned the same task id.
     #[test]
-    fn test_scheduler_task_ids() -> Result<()> {
+    fn from_task_id_returns_none_for_non_existing_task_id() -> Result<()> {
```

Comment:
Could you add a Rust Doc comment for this test, like we have for other tests in this module?